data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/processor/KinesisRecordProcessor.java

```diff
@@ -28,55 +37,67 @@
     public static final String KINESIS_RECORD_PROCESSING_ERRORS = "recordProcessingErrors";
     public static final String KINESIS_CHECKPOINT_FAILURES = "checkpointFailures";
     public static final String KINESIS_STREAM_TAG_KEY = "stream";
+    private KinesisCheckpointerTracker kinesisCheckpointerTracker;
+    private final ExecutorService executorService;
+    private AtomicBoolean isStopRequested;
 
-    public KinesisRecordProcessor(Buffer<Record<Event>> buffer,
+    public KinesisRecordProcessor(final BufferAccumulator<Record<Event>> bufferAccumulator,
                                   final KinesisSourceConfig kinesisSourceConfig,
                                   final AcknowledgementSetManager acknowledgementSetManager,
                                   final PluginMetrics pluginMetrics,
-                                  final PluginFactory pluginFactory,
+                                  final KinesisRecordConverter kinesisRecordConverter,
                                   final StreamIdentifier streamIdentifier) {
         this.bufferTimeoutMillis = (int) kinesisSourceConfig.getBufferTimeout().toMillis();
         this.streamIdentifier = streamIdentifier;
         this.kinesisSourceConfig = kinesisSourceConfig;
         this.kinesisStreamConfig = getStreamConfig(kinesisSourceConfig);
-        final PluginModel codecConfiguration = kinesisSourceConfig.getCodec();
-        final PluginSetting codecPluginSettings = new PluginSetting(codecConfiguration.getPluginName(), codecConfiguration.getPluginSettings());
-        this.codec = pluginFactory.loadPlugin(InputCodec.class, codecPluginSettings);
+        this.kinesisRecordConverter = kinesisRecordConverter;
         this.acknowledgementSetManager = acknowledgementSetManager;
-        this.acknowledgementSetCallbackCounter = pluginMetrics.counterWithTags(ACKNOWLEDGEMENT_SET_CALLBACK_METRIC_NAME, KINESIS_STREAM_TAG_KEY, streamIdentifier.streamName());
+        this.acknowledgementSetSuccesses = pluginMetrics.counterWithTags(ACKNOWLEDGEMENT_SET_SUCCESS_METRIC_NAME, KINESIS_STREAM_TAG_KEY, streamIdentifier.streamName());
+        this.acknowledgementSetFailures = pluginMetrics.counterWithTags(ACKNOWLEDGEMENT_SET_FAILURES_METRIC_NAME, KINESIS_STREAM_TAG_KEY, streamIdentifier.streamName());
+        this.recordsProcessed = pluginMetrics.counterWithTags(KINESIS_RECORD_PROCESSED, KINESIS_STREAM_TAG_KEY, streamIdentifier.streamName());
         this.recordProcessingErrors = pluginMetrics.counterWithTags(KINESIS_RECORD_PROCESSING_ERRORS, KINESIS_STREAM_TAG_KEY, streamIdentifier.streamName());
         this.checkpointFailures = pluginMetrics.counterWithTags(KINESIS_CHECKPOINT_FAILURES, KINESIS_STREAM_TAG_KEY, streamIdentifier.streamName());
         this.checkpointInterval = kinesisStreamConfig.getCheckPointInterval();
-        this.buffer = buffer;
+        this.bufferAccumulator = bufferAccumulator;
+        this.kinesisCheckpointerTracker = new KinesisCheckpointerTracker();
```

Comment:
Rather than include `setKinesisCheckpointTracker`, just pass this in the constructor. Then you can provide the mocks in the test using that. You can then remove the setter as this can lead to code bugs down the line. And you can make the field `final`.

Reply:
Thanks @dlvenable . I have removed the setter method and passed in the checkpoint tracker through the constructor.